src/web_task.cpp

```diff
@@ -41,8 +46,45 @@ void onWSEvent(AsyncWebSocket *server,
         Serial.printf("WebSocket client #%u disconnected\n", client->id());
         break;
     case WS_EVT_DATA:
-        Serial.printf("Raw json Message %.*s\n", len, data);
-        deserializeJson(input, data, len);
+    {
+        AwsFrameInfo * info = (AwsFrameInfo*)arg;
+        if(info->final && info->index == 0 && info->len == len)
+        {
+            // the whole message is in a single frame and we got all of it's data
+            Serial.printf("Raw json Message: %.*s\n", len, data);
+            deserializeJson(input, data, len);
+        }
+        else
+        {
+            // message is comprised of multiple frames or the frame is split into multiple packets
+			if (socketData == NULL) {
+				// allocate memory for buffer on first call
+				socketData  = (char *) malloc (SOCKET_DATA_SIZE);
+			}
+            for (size_t i = 0; i < len; i++)
+            {
+				// stop if message is bigger than buffer
+				if (currSocketBufferIndex >= SOCKET_DATA_SIZE - 1) {
+					Serial.printf("Raw json Message too big. Not processing.\n");
+					break;
+				}
+                // copy data of each chunk into buffer
+                socketData[currSocketBufferIndex] = data[i];
+                currSocketBufferIndex++;
+            }
+            if(info->final && currSocketBufferIndex >= info->len)
+            {
+				Serial.printf("Raw json Message: %.*s\n", currSocketBufferIndex, socketData);
+                // deserialize data after last chunk
+                socketData[currSocketBufferIndex] = '\0';
```

Comment:
is the zero-termination of the buffer needed? 
in the single frame variant we are also not terminating it. instead the equivalent would be to call deserializeJson with currSocketBufferIndex-1